pkg/frontend/querymiddleware/roundtrip.go

```diff
@@ -205,10 +207,17 @@ func newQueryTripperware(
 
 	queryInstantMiddleware := []Middleware{newLimitsMiddleware(limits, log)}
 
+	if cfg.SplitInstantQueriesByInterval > 0 {
+		queryInstantMiddleware = append(
+			queryInstantMiddleware,
+			newSplitInstantQueryByIntervalMiddleware(cfg.SplitInstantQueriesByInterval > 0, cfg.SplitInstantQueriesByInterval, limits, log, engine, registerer),
+			newInstrumentMiddleware("querysharding", metrics, log),
```

Comment:
This is incorrect here. It should moved back to its original place.

Reply:
🤦‍♀️ sorry about that

Reply:
Since this requires moving engine outside, should the `engineOpts.ActiveQueryTracker = nil` be also set for query splitting? Not sure what this parameter is used for